frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx

```diff
@@ -30,30 +36,41 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({ namespace })
     validateForm();
   };
 
+  const handleOnLoad = (items: { [key: string]: string }) => {
+    const isEmptyItems = _.isEmpty(items);
+    setIsZeroApplication(isEmptyItems);
+    if (isEmptyItems) {
+      setFieldValue('application.selectedKey', CREATE_APPLICATION_KEY);
+      setFieldValue('application.name', '');
+    }
+  };
+
   return (
     <React.Fragment>
-      <FormGroup
-        fieldId={fieldId}
-        label="Application"
-        helperTextInvalid={errorMessage}
-        isValid={isValid}
-        isRequired
-      >
-        <ApplicationDropdown
-          dropDownClassName="dropdown--full-width"
-          menuClassName="dropdown-menu--text-wrap"
-          id={fieldId}
-          namespace={namespace}
-          actionItem={{
-            actionTitle: 'Create New Application',
-            actionKey: CREATE_APPLICATION_KEY,
-          }}
-          autoSelect
-          selectedKey={selectedKey.value}
-          onChange={onDropdownChange}
-        />
-      </FormGroup>
-      {selectedKey.value === CREATE_APPLICATION_KEY && (
+      {!noProjectsAvailable && !isZeroApplication && (
```

Comment:
currently the ApplicationSelector has a react hook named `isZeroApplication` I changed it to `showAppDropdown` which takes in the `!noProjectsAvailable` as the initial value and I am using it to render Application Dropdown. This is all I could think of.

Reply:
As discussed, I'll review this and see if I can add anything else to the discussion.

Reply:
What you have now is great! :)

Reply:
@andrewballantyne  There's one issue with this, the first time the project and the application gets created successfully. Then if I go ahead and delete the Project and go back to the Git Import form the Application Dropdown shows up with the Loader. It is because the value of `noProjectsAvailable` is still false when the initial value of the local state is set and takes a moment to change to true but the state of `showAppDropdown` doesn't seem to update. The Application Dropdown shows up for quite sometime and requires continuous refresh.

Reply:
If I write it like
 ```suggestion
 const [isZeroApplication, setIsZeroApplication] = React.useState(false);
const showAppDropdown = !noProjectsAvailable && !isZeroApplication;
return(
<React.Fragment>
{showAppDropdown && (....``` the above issue disappears. Please let me know what you think

Reply:
If I write it like
 `const [isZeroApplication, setIsZeroApplication] = React.useState(false);
const showAppDropdown = !noProjectsAvailable && !isZeroApplication;
return(
<React.Fragment>
{showAppDropdown && (....`
the above issue disappears. Please let me know what you think

Reply:
Yeah, looks like that's what you'll have to do. It's not a big deal having that interim variable, I just would like to avoid a `not noValue` because it's a double negative and it was hard to read when I was first reviewing. Putting it into a variable helps for sure.